launcher/src/openvsix-registry.ts

```diff
@@ -0,0 +1,98 @@
+/**********************************************************************
+ * Copyright (c) 2023 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import { env } from "process";
+import * as fs from "./fs-extra";
+import { ProductJSON } from "./product-json";
+
+export class OpenVSIXRegistry {
+  /*****************************************************************************************************************
+   *
+   * Configures OpenVSIX registry.
+   *  - if OPENVSX_REGISTRY_URL environment variable exists, applies it as a extension registry.
+   *  - if not, then uses CHE_PLUGIN_REGISTRY_URL.
+   *
+   * It's a replacement of
+   * https://github.com/che-incubator/che-code/blob/f9389060f4dd7a435b13c75b63f6f12ec41fbd8e/build/scripts/entrypoint-volume.sh#L47-L66
+   *
+   *****************************************************************************************************************/
+  async configure(): Promise<void> {
+    console.log("# Configuring OpenVSIX registry...");
+
+    if (env.OPENVSX_REGISTRY_URL === undefined) {
+      console.log("  > env.OPENVSX_REGISTRY_URL is not set, skip this step");
+      return;
+    }
+
+    try {
+      let openvsxURL;
+      if (env.OPENVSX_REGISTRY_URL) {
+        console.log(
+          `  > env.OPENVSX_REGISTRY_URL set to ${env.OPENVSX_REGISTRY_URL}`
+        );
+        openvsxURL = `${env.OPENVSX_REGISTRY_URL}/vscode`;
```

Comment:
I think it makes sense to check what value we get from the env variable
let's say `https://open-vsx.org/` comes from the env variable
then there is a problem: URL contains double `/`:  `https://open-vsx.org//vscode`

I remember someone reported such problem...

Reply:
The same for CHE_PLUGIN_REGISTRY_URL.
I've added the verification for that, thanks!